app/src/main/java/com/example/amcc/view/MainFragment.java

```diff
@@ -72,9 +80,18 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
 
         navController = Navigation.findNavController(view);
         view.findViewById(R.id.btnResultID).setOnClickListener(v -> {
-            CarDetails car = new CarDetails(citySelected, Integer.parseInt(engineSizeField.getText().toString()), Integer.parseInt(emissionEdtField.getText().toString()), FuelType.e5, regDate, Integer.parseInt(avgConField.getText().toString()), Integer.parseInt(milePerYField.getText().toString()));
-            viewModel.setApiData(car);
-            navController.navigate(R.id.resultFragment);
+
+            validateInput();
+            if (validation.validate()) {
```

Comment:
@Jaber-aldroubi  here is duplicated code. Why do you have tow times validation? validateInput() could been return method. you should use one of them?

Reply:
validation is an object from Validation Class (Library ). validate() is a function in this class which will return (true or false) depends on validateInput, whichs return nothing at all.

Reply:
I mean this method should return validation.validate(); So you have to check if (validateInput())

Reply:
Reworked